raiden/channel/netting_channel.py

```diff
@@ -276,16 +277,19 @@ def get_settle_expiration(self, block_number):
 
         return blocks_until_settlement
 
-    def handle_closed(self, block_number):  # pylint: disable=unused-argument
+    def handle_closed(self, block_number, closing_address):  # pylint: disable=unused-argument
         balance_proof = self.our_state.balance_proof
 
         # the channel was closed, update our half of the state if we need to
-        closing_address = self.external_state.netting_channel.closing_address()
         if closing_address != self.our_state.address:
             self.external_state.update_transfer(balance_proof.balance_proof)
 
         unlock_proofs = balance_proof.get_known_unlocks()
-        self.external_state.withdraw(unlock_proofs)
+
+        try:
+            self.external_state.withdraw(unlock_proofs)
+        except AddressWithoutCode:
+            log.error('withdraw failed, channel is gone.')
```

Comment:
I'm also no 100% sure this is the best approach, I'm assuming that the node will pick up the event channel settled and cleanup after itself.